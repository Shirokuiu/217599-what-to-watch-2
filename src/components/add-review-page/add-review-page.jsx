import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {Header} from "../header/header";
import {AddReviewForm} from "../add-review-form/add-review-form";

import {withAddReviewForm} from "../../hocs/with-add-review-form/with-add-review-form";

const AddReviewFormWrapped = withAddReviewForm(AddReviewForm);

export const AddReviewPage = (props) => {
  const {match, movies, history} = props;
  const PARENT_STATUS = `ADD_REVIEW`;
  const {params} = match;
  const movieId = params.movieId - 1;
  const movie = movies[movieId];

  return <>
    {movies.length ? <section className="movie-card movie-card--full" style={{background: movie.backgroundColor}}>
      <div className="movie-card__header">
        <div className="movie-card__bg">
          <img src={movie.backgroundImage} alt={movie.name}/>
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <Header parent={PARENT_STATUS} movieId={movieId} movies={movies}/>

        <div className="movie-card__poster movie-card__poster--small">
          <img src={movie.posterImage} alt="The Grand Budapest Hotel poster" width="218" height="327"/>
        </div>
      </div>

      <div className="add-review">

        <AddReviewFormWrapped movieId={params.movieId} history={history}/>
      </div>

    </section> : null}
  </>;
};

AddReviewPage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      movieId: PropTypes.string
    })
  }),
  history: PropTypes.object.isRequired,
  movies: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    posterImage: PropTypes.string,
    previewImage: PropTypes.string,
    backgroundImage: PropTypes.string,
    backgroundColor: PropTypes.string,
    videoLink: PropTypes.string,
    previewVideoLink: PropTypes.string,
    description: PropTypes.string,
    rating: PropTypes.number,
    scoresCount: PropTypes.number,
    director: PropTypes.string,
    starring: PropTypes.array.string,
    runTime: PropTypes.number,
    genre: PropTypes.string,
    released: PropTypes.number,
    isFavorite: PropTypes.bool,
  })).isRequired
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  movies: state.movies
});

export default connect(mapStateToProps)(AddReviewPage);
