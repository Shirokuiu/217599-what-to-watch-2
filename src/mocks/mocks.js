export const films = [
  {
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts`
  },
  {
    img: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`
  },
  {
    img: `img/macbeth.jpg`,
    title: `Macbeth`
  },
  {
    img: `img/aviator.jpg`,
    title: `Aviator`
  },
  {
    img: `img/we-need-to-talk-about-kevin.jpg`,
    title: `We need to talk about Kevin`
  }
];
