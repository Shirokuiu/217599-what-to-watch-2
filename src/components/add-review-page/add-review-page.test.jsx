import React from "react";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import {AddReviewPage} from "./add-review-page";

import {films} from "../../mocks/mocks";

Enzyme.configure({adapter: new Adapter()});

it(`Is add review page rendered`, () => {
  const tree = shallow(<AddReviewPage
    match={{
      params: {
        movieId: `1`
      }
    }}
    history={{}}
    movies={films}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
