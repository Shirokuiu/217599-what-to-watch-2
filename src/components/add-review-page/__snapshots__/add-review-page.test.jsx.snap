// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is add review page rendered 1`] = `
<Fragment>
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "background": "#ffffff",
      }
    }
  >
    <div
      className="movie-card__header"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="The Grand Budapest Hotel"
          src="img/the-grand-budapest-hotel-bg.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <Header
        movieId={0}
        movies={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
              "director": "Wes Andreson",
              "genre": "Comedy",
              "id": 1,
              "isFavorite": false,
              "name": "The Grand Budapest Hotel",
              "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
              "previewImage": "img/the-grand-budapest-hotel.jpg",
              "previewVideoLink": "https://some-link",
              "rating": 8.9,
              "released": 2014,
              "runTime": 99,
              "scoresCount": 240,
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
              "videoLink": "https://some-link",
            },
          ]
        }
        parent="ADD_REVIEW"
      />
      <div
        className="movie-card__poster movie-card__poster--small"
      >
        <img
          alt="The Grand Budapest Hotel poster"
          height="327"
          src="img/the-grand-budapest-hotel-poster.jpg"
          width="218"
        />
      </div>
    </div>
    <div
      className="add-review"
    >
      <Connect(WithAddReviewForm)
        history={Object {}}
        movieId="1"
      />
    </div>
  </section>
</Fragment>
`;
